command-expansion-server/src/app.ts

```diff
@@ -36,41 +37,77 @@ app.post("/put-dictionary", async (req, res) => {
   console.log(`command-lib generated - path: ${commandDictionaryPath}`);
 
   const sqlExpression = `
-      insert into command_dictionary (command_types, mission, version)
-      values ($1, $2, $3)
-      on conflict (mission, version) do update
-      set command_types = $1
-      returning id;
-    `;
+    insert into command_dictionary (command_types, mission, version)
+    values ($1, $2, $3)
+    on conflict (mission, version) do update
+    set command_types = $1
+    returning id;
+  `;
 
   const { rows } = await db.query(sqlExpression, [
     commandDictionaryPath,
     parsedDictionary.header.mission_name,
     parsedDictionary.header.version,
   ]);
 
-  if (rows.length < 0) {
+  if (rows.length < 1) {
     console.error(`POST /dictionary: No command dictionary was updated in the database`);
-    res.status(400).json({ message: `POST /dictionary: No command dictionary was updated in the database` });
-  } else {
-    const id = rows[0].id;
-    res.status(200).json({ id });
+    res.status(500).send(`POST /dictionary: No command dictionary was updated in the database`);
+    return;
   }
+  const id = rows[0].id;
+  res.status(200).json({ id });
   return;
 });
 
-app.put("/expansion/:activityTypeName", async (req, res) => {
-  res.status(501).send("PUT /expansion: Not implemented");
+app.post('/get-expansion', async (req, res) => {
+  const expansionId = req.body.input.expansionId as string;
+  console.log(`POST /get-expansion: Retrieved expansion from database: id=${expansionId}`);
+
+  const { rows } = await db.query(`
+    SELECT expansion_logic
+    FROM expansion_rules
+    WHERE id = $1;
+  `, [
+    expansionId,
+  ]);
+
+  if (rows.length < 1) {
+    throw new ErrorWithStatusCode(`POST /get-expansion: No expansion with id=${expansionId}`, 404);
+  }
+
+  const expansionLogicTypescriptBase64 = rows[0].expansion_logic;
+  res.status(200).json({
+    typescript: expansionLogicTypescriptBase64,
+  });
   return;
 });
 
-app.get("/expansion/:expansionId(\\d+)", async (req, res) => {
-  res.status(501).send("GET /expansion: Not implemented");
+app.post('/put-expansion', async (req, res) => {
+  const activityTypeName = req.body.input.activityTypeName as string;
+  const expansionLogicBase64 = req.body.input.expansionLogic as string;
+
+  const { rows } = await db.query(`
+    INSERT INTO expansion_rules (activity_type, expansion_logic)
+    VALUES ($1, $2)
+    RETURNING id;
+  `, [
+    activityTypeName,
+    expansionLogicBase64,
+  ]);
+
+  if (rows.length < 1) {
+    throw new Error(`POST /put-expansion: No expansion was updated in the database`);
+  }
+
+  const expansionId = rows[0].id;
+  console.log(`POST /put-expansion: Updated expansion in the database: id=${expansionId}`);
+  res.status(200).json({ expansionId });
```

Comment:
Do we want the return to be `expansionId` or just `id`?

No strong preference, but I figured I'd pose the question

Reply:
I opted for `expansionId` just to be super clear, can be swayed to `id` though.

It helped me a bit to see the same name (when greping/highlighting) as this block:
```gql
type AddCommandExpansionResponse {
  expansionId: Int!
}
```

Reply:
I'm inclined to say let's go with id for three reasons. 

1. We have precedent for it here https://github.com/NASA-AMMOS/aerie/blob/develop/deployment/hasura/metadata/actions.graphql#L110-L112
2. The context of WHAT kind of id is being returned is implicated by the type its being returned in
3. If we are going to allow for Hasura to grab more information about them from the endpoint (ie return not just the id but the expansion as a complete type { id, typescriptCodeBase64 }) then id is more appropriate (many programs key off of typename + id as a caching key)

Reply:
For sure, just pushed this change!